tests/conftest.py

```diff
@@ -26,17 +32,114 @@ def app_params(request):
             kwargs.update(info.kwargs)
 
     args = [pargs[i] for i in sorted(pargs.keys())]
-    return args, kwargs
+
+    # ##### process pytest.mark.test_params
+
+    if test_params['specific_srcdir'] and 'srcdir' not in kwargs:
```

Comment:
What is difference between `srcdir` and `specific_srcdir`?
And it seems no testcases use the `specific_srcdir` parameter. Would you like to use it in next PR?